packages/ui/src/modal/editListing/EditListingModalRenderer.tsx

```diff
@@ -0,0 +1,441 @@
+
+  useEffect(() => {
+    setMarketplaces(
+      marketplaces.filter(
+        (marketplace) => marketplace.orderbook === 'reservoir'
+      )
+    )
+  }, [marketplaces.length])
+
+  useEffect(() => {
+    if (!open) {
+      setEditListingStep(EditListingStep.Edit)
+      setTransactionError(null)
+      setStepData(null)
+      setExpirationOption(expirationOptions[5])
+      setQuantity(1)
+    }
+  }, [open])
+
+  const { data: tokens } = useTokens(
+    open && {
+      tokens: [`${contract}:${tokenId}`],
+      includeAttributes: true,
+      normalizeRoyalties,
+    },
+    {
+      revalidateFirstPage: true,
+    }
+  )
+
+  const token = tokens && tokens.length > 0 ? tokens[0] : undefined
+  const is1155 = token?.token?.kind === 'erc1155'
+
+  const { data: userTokens } = useUserTokens(
+    open && is1155 ? account.address : undefined,
+    {
+      tokens: [`${contract}:${tokenId}`],
+    }
+  )
+
+  const quantityAvailable =
+    is1155 && userTokens[0]
+      ? Number(userTokens[0].ownership?.tokenCount || 1)
+      : 1
+
+  const chainCurrency = useChainCurrency()
+
+  const { data: onChainRoyalties } = useOnChainRoyalties({
+    contract,
+    tokenId,
+    chainId: chainCurrency.chainId,
+    enabled: enableOnChainRoyalties && open,
+  })
+
+  const onChainRoyaltyBps = useMemo(() => {
+    const totalRoyalty = onChainRoyalties?.[1].reduce((total, royalty) => {
+      total += parseFloat(formatUnits(royalty, currency?.decimals))
+      return total
+    }, 0)
+    if (totalRoyalty) {
+      return (totalRoyalty / 1) * 10000
+    }
+    return 0
+  }, [onChainRoyalties, chainCurrency])
+
+  if (enableOnChainRoyalties && onChainRoyaltyBps) {
+    royaltyBps = onChainRoyaltyBps
+  }
+
+  const editListing = useCallback(() => {
+    if (!signer) {
+      const error = new Error('Missing a signer')
+      setTransactionError(error)
+      throw error
+    }
+
+    if (!client) {
+      const error = new Error('ReservoirClient was not initialized')
+      setTransactionError(error)
+      throw error
+    }
+
+    if (!listingId) {
+      const error = new Error('Missing list id to edit')
+      setTransactionError(error)
+      throw error
+    }
+
+    if (!isOracleOrder) {
+      const error = new Error('Not an oracle order')
+      setTransactionError(error)
+      throw error
+    }
+
+    setTransactionError(null)
+
+    let expirationTime: string | null = null
+
+    if (expirationOption.relativeTime && expirationOption.relativeTimeUnit) {
+      expirationTime = dayjs()
+        .add(expirationOption.relativeTime, expirationOption.relativeTimeUnit)
+        .unix()
+        .toString()
+    }
+
+    let reservoir = marketplaces[0]
+
+    const listing: Listings[0] = {
+      token: `${contract}:${tokenId}`,
+      weiPrice: parseUnits(`${price}`, currency?.decimals)
+        .mul(quantity)
+        .toString(),
+      //@ts-ignore
+      orderbook: reservoir.orderbook,
```

Comment:
we can simplify this here because the orderbook will always be "reservoir" and the orderKind will always be seaport-v1.4